### ISSUE TITLE
Action fails after 1h+ despite successful deployment

### ISSUE THREAD

**Issue by JacopoMadaluni** [Azure/webapps-deploy#444]
Hello,
since a few months ago or so, the `azure/webapps-deploy@v3` fails after 1h+ despite a successful deployment.
After this failure, I usually need to restart the app service in order for the changes to go through which isn't ideal.
Whilst this feels like a broader app service issue, I am not sure where to report it as it only happens when I use this action.

My usage in CI is as follows:
```
            - name: "Deploy to Azure Web App"
              id: deploy-to-webapp
              uses: azure/webapps-deploy@v3
              with:
                  app-name: "${{ env.APP_NAME }}"
                  slot-name: "production"
                  publish-profile: "${{ env.PUBLISH_PROFILE }}"
                  package: .
```

This is the error log from the action
![Image](https://github.com/user-attachments/assets/2219b892-735d-43a1-be0b-d42836c7ae23)
```
Package deployment using OneDeploy initiated.
{
  id: '2b592529-1c8f-47da-8110-e9274d94f874',
  status: 3,
  status_text: 'Deployment has been stopped due to SCM container restart. The restart can happen due to a management operation on site. Do not perform a management operation and a deployment operation in quick succession. Adding a small delay can help avoid any conflicts.',
  author_email: 'N/A',
  author: 'N/A',
  deployer: 'OneDeploy',
  message: 'OneDeploy',
  progress: '',
  received_time: '2024-10-02T11:43:17.2412909Z',
  start_time: '2024-10-02T11:43:21.0833868Z',
  end_time: '2024-10-09T18:31:10.8607467Z',
  last_success_end_time: null,
  complete: true,
  active: false,
  is_temp: false,
  is_readonly: true,
  url: '***',
  log_url: '***,
  site_name: ***',
  build_summary: { errors: [], warnings: [] }
}
Updating submodules.
Preparing deployment for commit id '2b592529-1'.
PreDeployment: context.CleanOutputPath False
PreDeployment: context.OutputPath /home/site/wwwroot
Repository path is /tmp/zipdeploy/extracted
Running oryx build...
Command: oryx build /tmp/zipdeploy/extracted -o /home/site/wwwroot --platform nodejs --platform-version 18 -p virtualenv_name= --log-file /tmp/build-debug.log  -i /tmp/8dce2d76aa4ba8f -p compress_node_modules=tar-gz | tee /tmp/oryx-build.log
Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20240424.1, Commit: d37b2225a252ab2c04b4726024d047cf01ea[13](https://github.com/***/actions/runs/11259972295/job/31310153804#step:12:14)18, ReleaseTagName: 20240424.1

Build Operation ID: ce22f881a5c78847
Repository Commit : 2b592529-1c8f-47da-8110-e9274d94f874
OS Type           : bullseye
Image Type        : githubactions

Detecting platforms...
Error: Failed to deploy web package to App Service.
Error: Deployment Failed, Package deployment using ZIP Deploy failed. Refer logs for more details.
App Service Application URL: http://***.azurewebsites.netPackage deployment using OneDeploy initiated.
{
  id: '2b592529-1c8f-47da-8110-e9274d94f874',
  status: 3,
  status_text: 'Deployment has been stopped due to SCM container restart. The restart can happen due to a management operation on site. Do not perform a management operation and a deployment operation in quick succession. Adding a small delay can help avoid any conflicts.',
  author_email: 'N/A',
  author: 'N/A',
  deployer: 'OneDeploy',
  message: 'OneDeploy',
  progress: '',
  received_time: '2024-10-02T11:43:17.2412909Z',
  start_time: '2024-10-02T11:43:21.0833868Z',
  end_time: '2024-10-09T18:31:10.8607467Z',
  last_success_end_time: null,
  complete: true,
  active: false,
  is_temp: false,
  is_readonly: true,
  url: '***',
  log_url: '***',
  site_name: '***',
  build_summary: { errors: [], warnings: [] }
}
Updating submodules.
Preparing deployment for commit id '2b592529-1'.
PreDeployment: context.CleanOutputPath False
PreDeployment: context.OutputPath /home/site/wwwroot
Repository path is /tmp/zipdeploy/extracted
Running oryx build...
Command: oryx build /tmp/zipdeploy/extracted -o /home/site/wwwroot --platform nodejs --platform-version 18 -p virtualenv_name= --log-file /tmp/build-debug.log  -i /tmp/8dce2d76aa4ba8f -p compress_node_modules=tar-gz | tee /tmp/oryx-build.log
Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20240424.1, Commit: d37b2225a252ab2c04b4726024d047cf01ea[13](***/actions/runs/11259972295/job/31310153804#step:12:14)18, ReleaseTagName: 20240424.1

Build Operation ID: ce22f881a5c78847
Repository Commit : 2b592529-1c8f-47da-8110-e9274d94f874
OS Type           : bullseye
Image Type        : githubactions

Detecting platforms...
Error: Failed to deploy web package to App Service.
Error: Deployment Failed, Package deployment using ZIP Deploy failed. Refer logs for more details.
App Service Application URL: ***
```

**Comment by copdips** [comment]
When reviewing the debug logs, it appears that the system is checking the wrong deployment ID, rather than the one just deployed.

According to the action logs, the deployment ID is `4ec00f60-6727-4255-a7ed-ceee3c24294b`, which corresponds to a failed deployment from five hours ago. Since this incorrect deployment failed, its status remains stuck at `Building and Deploying`.

I suspect there may be an issue at the Azure WebApp API level.
Tested locally with `az webapp deploy`, same issue.

![Image](https://github.com/user-attachments/assets/703bc8a8-9181-43ed-ac9e-dec3ca8cee7c)

![Image](https://github.com/user-attachments/assets/50bdd2b0-572e-4914-957c-8b788e3b70eb)

**Comment by ShilpiRachna1** [comment]
@JacopoMadaluni , @copdips - We are working on fixing this. Will share an update on ETA soon

**Comment by SteenMolberg** [comment]
I'm seeing exact the same as @copdips. I deleted a failed deployment from the deployment log on the webapp and the deployment went through.

**Comment by ShilpiRachna1** [comment]
We have a fix that should be available by end of March

**Comment by BenjaminAbt** [comment]
@ShilpiRachna1 could you please provide more information? 

We have the problem on some of our own and customer projects that the deployment fails because of exactly this error - repeatedly.
Our only solution is to replace App Service if we do not have a solution in a timely manner.

**Comment by stmaute** [comment]
@BenjaminAbt Did you read what I wrote regarding how I fixed it and got it working?

**Comment by BenjaminAbt** [comment]
@stmaute yes, but its not sustainable. Sometimes we cant even delete deployment logs.

.. logs run in timeouts

![Image](https://github.com/user-attachments/assets/0ca818dc-a54c-4645-9c3a-398d9d9ec09e)

or on delete we just get the error "Error: Error"

**Comment by stmaute** [comment]
I agree that this is not a sustainable solution but at least I got my deployment working again.
The root cause should really be fixed in near future.

**Comment by LGNorris** [comment]
@ShilpiRachna1 any updates of said potential fix for this issue?

**Comment by BenjaminAbt** [comment]
We have been able to partially solve our deployment problems by using a more powerful app service plan.\
That somehow makes absolutely no sense at first - but that is simply the result. We haven't had a single aborted deployment since then, regardless of whether we have an aborted deployment in our history or not. Maybe it really is the solution, maybe it's just a race condition with the rollout of the fix.

The bug handling information is absolutely disrespectful.

**Comment by bschickerbauer** [comment]
I am facing exact same issue: Deployments > 1+ hour do fail.
`    - uses: azure/webapps-deploy@v3
      with:
        app-name: ${{ env.AZURE_WEBAPP_NAME }}
`

From Az web app perspective, the deployment is successful and the logs proof that accordingly:
![Image](https://github.com/user-attachments/assets/503d4dc7-686b-40e2-a3bd-3b777bbdce53)

Though, after more than 1h+ the GH action does not seem to have a valid token anymore, nor is it able to request a new one. Hence, the action fails due to 403.
![Image](https://github.com/user-attachments/assets/ced13d7f-b541-44f4-bde6-ab51d30e723a)

Basically the gh action information is inaproppriate and misleading.
